tests/test_processing_common.py

```diff
@@ -0,0 +1,127 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import tempfile
+import unittest
+
+from transformers import CLIPTokenizerFast, ProcessorMixin
+from transformers.models.auto.processing_auto import processor_class_from_name
+from transformers.testing_utils import (
+    check_json_file_has_correct_format,
+    require_tokenizers,
+    require_torch,
+    require_vision,
+)
+from transformers.utils import is_vision_available
+
+
+if is_vision_available():
+    from transformers import CLIPImageProcessor
+
+
+@require_torch
+class ProcessorTesterMixin:
+    processor_class = None
+
+    def prepare_processor_dict(self):
+        return {}
+
+    def get_component(self, attribute, **kwargs):
+        assert attribute in self.processor_class.attributes
+        component_class_name = getattr(self.processor_class, f"{attribute}_class")
+        if isinstance(component_class_name, tuple):
+            component_class_name = component_class_name[0]
+
+        component_class = processor_class_from_name(component_class_name)
+        component = component_class.from_pretrained(self.tmpdirname, **kwargs)  # noqa
+
+        return component
+
+    def prepare_components(self):
+        components = {}
+        for attribute in self.processor_class.attributes:
+            component = self.get_component(attribute)
+            components[attribute] = component
+
+        return components
```

Comment:
naming conventions are a bit new? 
Would be nice to keep some equivalence between the equivalent config / tokenizer / feature extractors tests but anyway.

Reply:
the difference is that `config / tokenizer / feature extractors` don't have components, but processor has.